core/flay_service.cpp

```diff
@@ -191,6 +195,10 @@ int FlayServiceWrapper::run() {
             recordProgramChange(flayService);
             semanticsChangeCounter++;
         }
+        if (FlayOptions::get().getOptimizedOutputDir() != std::nullopt) {
+            std::filesystem::path fileName = "optimized." + std::to_string(updateIdx) + ".p4";
```

Comment:
You can use https://en.cppreference.com/w/cpp/filesystem/path/replace_extension here